src/diffdrive_roscore.cpp

```diff
@@ -80,10 +98,24 @@ namespace roboclaw {
         motor_vel.index = 0;
         motor_vel.mot1_vel_sps = 0;
         motor_vel.mot2_vel_sps = 0;
+        motor_vel.acceleration = 0;
 
-        // Linear
-        motor_vel.mot1_vel_sps += (int) (steps_per_meter * msg.linear.x);
-        motor_vel.mot2_vel_sps += (int) (steps_per_meter * msg.linear.x);
+        // Linear acceleration
+        motor_vel.acceleration = linear_acceleration;
+
+        // Linear speed
+        double linear_speed_x = msg.linear.x;
+        if(linear_speed_x > max_linear_speed){
+            linear_speed_x = max_linear_speed;
+            ROS_WARN_STREAM_THROTTLE(15, "Linear speed clipped at max speed of " << max_linear_speed << " m/s");
+        }
+        else if (linear_speed_x < -max_linear_speed){
+            linear_speed_x = -max_linear_speed;
+            ROS_WARN_STREAM_THROTTLE(15, "Linear speed clipped at min speed of " << -max_linear_speed << " m/s");
+        }
+
+        motor_vel.mot1_vel_sps += (int) (steps_per_meter * linear_speed_x);
+        motor_vel.mot2_vel_sps += (int) (steps_per_meter * linear_speed_x);
 
         if(msg.linear.y > 0){
```

Comment:
What happens if msg.linear.y < 0?

Reply:
There is a _else if_ at line 122. I think Github hides the following lines. You can expand the view to see them :)

Reply:
my bad!